core/src/test/java/org/projectnessie/cel/extension/StringsTest.java

```diff
@@ -0,0 +1,349 @@
+        testExpression(new TestData("'ta©o©αT'.lastIndexOf('©α', 4) == 4"));
+    */
+
+    // Error test cases based on checked expression usage.
+    /*
+        testExpression(
+            new TestData("'tacocat'.lastIndexOf('a', -1) == -1", "String index out of range: -1"));
+        testExpression(
+            new TestData("'tacocat'.lastIndexOf('a', 30) == -1", "String index out of range: 30"));
+    */
+
+    // Valid parse-only expressions which should generate runtime errors.
+    testExpression(new TestData("24.lastIndexOf('2') == 0", "no matching overload", true));
+    testExpression(new TestData("'hello'.lastIndexOf(true) == 1", "no matching overload", true));
+    testExpression(new TestData("42.lastIndexOf('4', 0) == 0", "no matching overload", true));
+    testExpression(new TestData("'42'.lastIndexOf(4, 0) == 0", "no matching overload", true));
+    testExpression(new TestData("'42'.lastIndexOf('4', '0') == 0", "no matching overload", true));
+    testExpression(new TestData("'42'.lastIndexOf('4', 0, 1) == 0", "no matching overload", true));
+  }
+
+  @Test
+  public void testLowerAscii() {
+    testExpression(new TestData("'TacoCat'.lowerAscii() == 'tacocat'"));
+    testExpression(new TestData("'TacoCÆt Xii'.lowerAscii() == 'tacocÆt xii'"));
+  }
+
+  @Test
+  public void testReplace() {
+    testExpression(new TestData("'hello hello'.replace('he', 'we') == 'wello wello'"));
+
+    testExpression(
+        new TestData("\"12 days 12 hours\".replace(\"{0}\", \"2\") == \"12 days 12 hours\""));
+    testExpression(
+        new TestData("\"{0} days {0} hours\".replace(\"{0}\", \"2\") == \"2 days 2 hours\""));
+    /*
+        testExpression(new TestData("\"{0} days {0} hours\".replace(\"{0}\", \"2\", 1).replace(\"{0}\", \"23\") == \"2 days 23 hours\""));
+    */
+    testExpression(new TestData("\"1 ©αT taco\".replace(\"αT\", \"o©α\") == \"1 ©o©α taco\""));
+
+    // Valid parse-only expressions which should generate runtime errors.
+    testExpression(new TestData("42.replace(2, 1) == \"41\"", "no matching overload", true));
+    testExpression(new TestData("\"42\".replace(2, 1) == \"41\"", "no matching overload", true));
+    testExpression(
+        new TestData("\"42\".replace(\"2\", 1) == \"41\"", "no matching overload", true));
+    testExpression(
+        new TestData("42.replace(\"2\", \"1\", 1) == \"41\"", "no matching overload", true));
+    testExpression(
+        new TestData("\"42\".replace(2, \"1\", 1) == \"41\"", "no matching overload", true));
+    testExpression(
+        new TestData("\"42\".replace(\"2\", 1, 1) == \"41\"", "no matching overload", true));
+    testExpression(
+        new TestData(
+            "\"42\".replace(\"2\", \"1\", \"1\") == \"41\"", "no matching overload", true));
+    testExpression(
+        new TestData(
+            "\"42\".replace(\"2\", \"1\", 1, false) == \"41\"", "no matching overload", true));
+  }
+
+  @Test
+  public void testSplit() {
```

Comment:
Looks like there are lot of duplicate tests from other test functions in this one.